Cohorted Courseware: Icon Syntax Sync (TNL-1137)

talbs opened the pull request:
This work syncs icon class syntax (to the new Font Awesome conventions) + semantically hides icon from assistive tech.

talbs commented:
@andy-armstrong and @clrux, please review.

 - - -

FYI, @cahrens and @explorerleslie.

clrux commented on cms/templates/ux/reference/modal_access-component.html:
```diff
@@ -13,8 +13,8 @@ <h2 class="title modal-window-title">Editing visibility for: [Component Name]</h
-                <i class="icon icon-warning-sign"></i>
-                <p class="copy">This component is contained in a unit that is hidden from students. Component visibility settings are overridden by the unit visibility settings.</p>
+                <i class="icon fa fa-exclamation-triangle" aria-hidden="true"></i>
+                <p class="copy"><span class="sr">Warning: </span>This component is contained in a unit that is hidden from students. Component visibility settings are overridden by the unit visibility settings.</p>
```
Nicely done.

cptvitamin replied:
Like a boss, really.

cahrens commented:
@talbs please create a sandbox

clrux commented on lms/templates/instructor/instructor_dashboard_2/cohort-form.underscore:
```diff
@@ -71,14 +71,14 @@
                             <div class="input-group-other">
                                 <div class="msg-inline">
-                                    <p class="copy-error"><i class="icon icon-warning-sign"></i><%- gettext("No content groups exist. Create a content group to associate with cohort groups.") %> <a class="link-to-group-settings" href="<%- studioGroupConfigurationsUrl %>"><%- gettext("Create a content group") %></a></p>
+                                    <p class="copy-error"><i class="icon fa fa-exclamation-triangle" aria-hidden="true"></i><%- gettext("No content groups exist. Create a content group to associate with cohort groups.") %> <a class="link-to-group-settings" href="<%- studioGroupConfigurationsUrl %>"><%- gettext("Create a content group") %></a></p>
```
Maybe add `.sr` text for "Warning:" like you did above?

clrux commented on lms/templates/instructor/instructor_dashboard_2/cohort-form.underscore:
```diff
@@ -71,14 +71,14 @@
                                 <% if (hasSelectedContentGroup && !foundSelected) { %>
                                     <div class="msg-inline">
-                                        <p class="copy-error"><i class="icon icon-warning-sign"></i><%- gettext("The previously selected content group was deleted. Select another content group.") %></p>
+                                        <p class="copy-error"><i class="icon fa fa-exclamation-triangle" aria-hidden="true"></i><%- gettext("The previously selected content group was deleted. Select another content group.") %></p>
```
Maybe add `.sr` text for "Warning:" like you did above?

talbs commented:
@clrux, addressed your comments, thanks.

@cahrens, sandbox building now. It'll be available at cohorted-courseware-iconz.m.sandbox.edx.org shortly.

andy-armstrong commented:
@talbs This should fix all the issues I was seeing, although I'd like to double check on the sandbox. Thanks for jumping on this. :+1:

cahrens commented on cms/templates/visibility_editor.html:
```diff
@@ -27,8 +27,8 @@ <h4 class="title">${_('You have not set up any groups to manage visibility with.
-            <i class="icon icon-warning-sign"></i>
-            <p class="copy">${_('The Unit this component is contained in is hidden from students. Visibility settings here will be trumped by this.')}</p>
+            <i class="icon fa fa-exclamation-triangle" aria-hidden="true"></i>
+            <p class="copy"><span class='sr'>${_('Warning: ')}</span>${_('The Unit this component is contained in is hidden from students. Visibility settings here will be trumped by this.')}</p>
```
Some notes to the translators would be good here. It is important that "Warning:" have a trailing space.

A better way to do this would be with placeholders to keep the HTML out of the translatable string, but allow the translators to deal with the string as a whole. For example,
"{span_tag_start}Warning: {span_tag_end} The Unit this component...".format

cahrens commented:
And note that some tests will need to be updated.

talbs commented:
Rebased and handing off to @andy-armstrong for help with tests and final i8tn syntax tweaks on https://github.com/edx/edx-platform/commit/63c2d1145d7431f5417f8a26b208eac0b1ae62c8

andy-armstrong commented:
Everything looks great on the sandbox. I'm going to work on cleaning up the i18n and tests.

talbs commented:
FYI, @cptvitamin. per your feature feedback (https://github.com/edx/edx-platform/pull/6299/files#r22548500), I hid some things from the aria monster here. :)

andy-armstrong commented:
I've fixed the i18n and tests and I've squashed the commits. I'm just awaiting the unit test build on Jenkins to see if everything passed (it did locally). 

@clrux @cahrens - please re-review when you can so we can get this merged.

cahrens commented on cms/templates/visibility_editor.html:
```diff
@@ -27,8 +27,17 @@ <h4 class="title">${_('You have not set up any groups to manage visibility with.
+            <p class="copy">
+                ## Translators: The "Warning:" text is meant to communicate the type of message being communicated and should always end with an extra space
+                ${_(
+                    "{span_tag_start}Warning: {span_tag_end} The Unit this component is contained in is hidden from students. Visibility settings here will be trumped by this."
```
@andy-armstrong There are two spaces now between "Warning:" and "The".

cahrens commented on cms/templates/visibility_editor.html:
```diff
@@ -27,8 +27,17 @@ <h4 class="title">${_('You have not set up any groups to manage visibility with.
-            <p class="copy">${_('The Unit this component is contained in is hidden from students. Visibility settings here will be trumped by this.')}</p>
+            <i class="icon fa fa-exclamation-triangle" aria-hidden="true"></i>
+            <p class="copy">
+                ## Translators: The "Warning:" text is meant to communicate the type of message being communicated and should always end with an extra space
```
Now that the message is all in one string, this translator note probably isn't needed. But if you think it is still helpful, it is fine to leave it in.

andy-armstrong replied:
@cahrens I didn't write this one (I think DB helped @talbs). Since it got merged I'll make your suggested change on cohorted-courseware with my doc changes:

https://github.com/edx/edx-platform/pull/6498

cahrens commented on lms/static/js/spec/instructor_dashboard/ecommerce_spec.js:
```diff
@@ -20,7 +20,7 @@ define(['backbone', 'jquery', 'js/instructor_dashboard/ecommerce', 'js/common_he
             });
 
-            it("shows the input field when the checkbox is checked", function () {
+            it("shows the input fielmd when the checkbox is checked", function () {
```
typo (in an unrelated file...)

cahrens commented:
:+1: after you fix the minor issues I noted.

talbs commented:
Closing this since the work/commits are captured in https://github.com/edx/edx-platform/pull/6539. Thanks for the assists and reviews, @cahrens and @andy-armstrong!

cahrens commented:
@andy-armstrong This PR wasn't merged-- @talbs closed it. I don't understand why though.

andy-armstrong commented:
@cahrens It looks like @talbs was confused by the temporary branch I created from this to create the manual testing sandbox. I've reopened this, I'll make the fix, and then I'll merge.

andy-armstrong commented:
@cahrens I've pushed up fixes to address your feedback. I've squashed them too since you gave me a thumbs up, so I'll merge once I get a clean build.
